### PR TITLE
Add premium domain tracking

### PR DESCRIPTION
**Pull request by hambai**
Recently we introduced premium domain purchases in Calypso but we never added any tracking to those so we don't know how often those are suggested or clicked on so here's a small update to fix that.

#### Changes proposed in this Pull Request

* ~Add railcar suffix for premium domains~
* Add is_premium property to the `calypso_domain_search_add_button_click` tracks event
* Add `domain_type` property to the traintracks event

#### Testing instructions

* Verify that when premium domains are rendered there is a `#premium` railcar suffix
* Verify that when you click on a premium domain you'll get `is_premium` set to `true` in `calypso_domain_search_add_button_click`

### CODE REVIEW

File: client/components/domains/domain-registration-suggestion/index.jsx [hunk 1]
```diff
@@ -81,7 +81,9 @@ class DomainRegistrationSuggestion extends React.Component {
 	recordRender() {
 		if ( this.props.railcarId && isNumber( this.props.uiPosition ) ) {
 			let resultSuffix = '';
-			if ( this.props.suggestion.isRecommended ) {
+			if ( this.props.suggestion.is_premium ) {
```
Comment (daledupreez) [inline client/components/domains/domain-registration-suggestion/index.jsx:84]: @hambai, this construction will mask cases where the premium domain appears in position 1 or 2 (aka when the domain is recommended or the best alternative).

I think it would be better to leave the existing `if/else if` logic and add another `if` block to capture this.

Comment (hambai) [reply client/components/domains/domain-registration-suggestion/index.jsx:84]: huh yeah - I've asked our data engineering team for help as to where/how to store that info

Comment (hambai) [reply client/components/domains/domain-registration-suggestion/index.jsx]: After discussing this internally - p1601461479028900-slack-C03DKP1JP - we're going with additional suffix (not replacing the current suffix)

File: client/components/domains/domain-registration-suggestion/index.jsx [hunk 2]
```diff
@@ -87,6 +87,10 @@ class DomainRegistrationSuggestion extends React.Component {
 				resultSuffix = '#best-alternative';
 			}
 
+			if ( this.props.suggestion.is_premium ) {
+				resultSuffix = resultSuffix + '#premium';
```
Comment (klimeryk) [inline client/components/domains/domain-registration-suggestion/index.jsx:91]: Super nitpick, but I prefer `resultSuffix += '#premium';`. Otherwise, I'm suspicious that the second `resultSuffix` is actually somehow different than the first one - since there must _have_ been a reason for not using the `+=` operator!
TL;DR: it's shorter, easier to parse (for human) and less prone to errors/typos 🙇

**Review by klimeryk — APPROVED** [review]
Tested with a regular and premium domains and LGTM!

File: client/components/domains/domain-registration-suggestion/index.jsx (hunk 2 above)
Comment (hambai) [reply client/components/domains/domain-registration-suggestion/index.jsx]: Well - it turns our we can add a separate custom property so no need for a suffix :D
